### PR TITLE
fix: fb custom audience html response

### PR DESCRIPTION
**Pull request by aashishmalik**
## What are the changes introduced in this PR?

Write a brief explainer on your code changes.

## What is the related Linear task?

Resolves INT-2015

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3402/test-report.html

File: src/v0/util/facebookUtils/index.test.js [hunk 1]
```diff
@@ -639,3 +640,25 @@ describe('getContentType', () => {
     expect(result).toBe(defaultValue);
   });
 });
+
+describe('isHtmlFormat', () => {
```
Comment (sanpj2292) [inline src/v0/util/facebookUtils/index.test.js:644]: Can we add a test-case for undefined as well ?
`isHtmlFormat(undefined)`

Comment (aashishmalik) [reply src/v0/util/facebookUtils/index.test.js:644]: added few extra TC as well

File: test/integrations/destinations/fb_custom_audience/network.ts [hunk 2]
```diff
@@ -523,4 +523,122 @@ export const networkCallsData = [
...
+      body: {
+        JSON: {},
+        XML: {},
+        JSON_ARRAY: {},
+        FORM: {},
+      },
+      files: {},
+    },
+    httpRes:
+      '<!DOCTYPE html><html> <body> <h1>My First Heading</h1><p>My first paragraph.</p> </body></html>',
```
Comment (sanpj2292) [inline test/integrations/destinations/fb_custom_audience/network.ts:642]: Which test-case is using this mock ?

Comment (aashishmalik) [reply test/integrations/destinations/fb_custom_audience/network.ts:642]: added testcase for this in other.ts

Comment (aashishmalik) [reply test/integrations/destinations/fb_custom_audience/network.ts:642]: this is replica of fb aud response

File: src/v0/util/facebookUtils/index.test.js [hunk 3]
```diff
@@ -639,3 +640,41 @@ describe('getContentType', () => {
...
+
+  it('should return false for null', () => {
+    expect(isHtmlFormat(null)).toBe(false);
+  });
+
+  it('should return false for empty string', () => {
+    expect(isHtmlFormat('')).toBe(false);
+  });
+
+  it('should return false for html doctype', () => {
```
Comment (sanpj2292) [inline src/v0/util/facebookUtils/index.test.js:673]: This test-case seems to be mis-leading. Can you check ?

File: src/v0/util/facebookUtils/index.js [hunk 4]
```diff
@@ -292,7 +292,13 @@ const formingFinalResponse = (
   throw new TransformationError('Payload could not be constructed');
 };
 
+const isHtmlFormat = (string) => {
+  const pattern = /<([a-z]+)\b[^>]*>(.*?)<\/\1>/i;
```
Comment (sanpj2292) [inline src/v0/util/facebookUtils/index.js:296]: This regex would even validate if tomorrow the integration sends xml response right ?!

Comment (aashishmalik) [reply src/v0/util/facebookUtils/index.js:296]: regex should be able to identify XML format also

Comment (aashishmalik) [reply src/v0/util/facebookUtils/index.js:296]: it('should return true for xml doctype', () => {
    expect(
      isHtmlFormat(
        '"<note><to>Tove</to><from>Jani</from><heading>Reminder</heading><body>Don\'t forget me this weekend!</body></note>"',
      ),
    ).toBe(true);
  });
  
  this case passed

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/util/facebookUtils/index.js (hunk 4 above)
Comment (sanpj2292) [reply src/v0/util/facebookUtils/index.js:296]: If that is the case, then this function should be renamed to something else. As others might use it thinking it is for HTML only but this function identifies XML as well.
(Or)
Update the regex to identify only HTML & discard any other kind of output(XML)

Comment (aashishmalik) [reply src/v0/util/facebookUtils/index.js]: done to identify html only

**Review by sanpj2292 — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

**Pull request merged**